### PR TITLE
tree-wide: update JUnit AUnit output for Skipped tests

### LINKED ISSUE #101: Proper encoding of skipped AUNIT tests into junit results xml
The current implementation encodes skipped tests as:
```xml
<testcase classname="net.cars.engine.ValveTest" name="moveDown" status="SKIP">
  <system-err>Test execution skipped due to missing prerequisites in car engine</system-err>
  <error type="abortion" message="Missing Prerequisites - Skipped due to potential obsolete</error>
</testcase>
```
It would be great to align with aunit schema/standard:

```xml
<testcase classname="net.cars.engine.ValveTest" name="moveDown">
    <skipped />
</testcase>
```

or with message:

```xml
<testcase classname="net.cars.engine.ValveTest" name="moveDown">
    <skipped message="Test execution skipped due to missing prerequisites in car engine"/>
</testcase>
```

More details regarding junit format:
https://llg.cubic.org/docs/junit/

The relevant piece of the sapcli code seems to be here:
https://github.com/jfilak/sapcli/blob/master/sap/cli/aunit.py#L275

### PR DESCRIPTION
**Pull request by jfilak**
I was not sure how to convert ABAP ADT AUnit XML into JUnit.

I have decided to consider skipped only the tests with the kind ABORTION and the severity TOLERABLE - I consulted ABAP code of cl_abap_unit_assert=>skip( ).

Jenkins output:

    DO_THE_FAIL – ZCL_THEKING_MANUAL_HARDCORE=>LTCL_TEST

    Standard Output:
    Include: <ZCL_THEKING_MANUAL_HARDCORE===CCAU> Line: <19> (DO_THE_FAIL)

    Standard Error:
    Test execution skipped due to missing prerequisites
    Test 'LTCL_TEST->DO_THE_FAIL' in Main Program 'ZCL_THEKING_MANUAL_HARDCORE===CP'.

Closes #101

### CODE REVIEW

File: sap/adt/aunit.py [hunk 1]
```diff
@@ -297,6 +303,14 @@ def is_warning(self):
 
         return self.severity == AlertSeverity.TOLERABLE
 
+    @property
+    def is_skipp(self):
```
Comment (mnezerka) [inline sap/adt/aunit.py:307]: Why `is_skipp` instead of `is_skipped`?

File: sap/adt/aunit.py [hunk 2]
```diff
@@ -297,6 +303,14 @@ def is_warning(self):
...
+        """Returns true if the alert a message about skipped test.
+           ABAP: cl_abap_unit_assert=>skip( 'Skipped because of a wrong system' ).
+        """
+
+        return self.severity == AlertSeverity.TOLERABLE and self.kind == AlertKind.ABORTION
```
Comment (mnezerka) [inline sap/adt/aunit.py:312]: Why don't we mark skipped findings that are `not TOLERABLE`? Shouldn't we skip all findings that have kind equal to `abortion`? Does this logic come from `cl_abap_unit_assert=>skip` abap code mentioned in doc string?

File: test/unit/test_sap_cli_aunit.py [hunk 3]
```diff
@@ -281,6 +282,30 @@ def test_aunit_class_junit4_syntax_error(self):
...
+        self.connection.set_responses(Response(status_code=200, text=AUNIT_RESULTS_SKIPPED_XML, headers={}))
+
+        with patch('sap.cli.core.get_console', return_value=BufferConsole()) as mock_print:
+            retval = self.execute_run('class', 'yclass', '--output', 'junit4', '--result', ResultOptions.ONLY_UNIT.value)
+
+        self.maxDiff = None
+        self.assertEqual(mock_print.return_value.capout, '''<?xml version="1.0" encoding="UTF-8" ?>
+<testsuites name="yclass">
+  <testsuite name="LTCL_TEST" package="ZCL_THEKING_MANUAL_HARDCORE" tests="1">
+    <testcase name="DO_THE_FAIL" classname="ZCL_THEKING_MANUAL_HARDCORE=&gt;LTCL_TEST" status="SKIP">
```
Comment (mnezerka) [inline test/unit/test_sap_cli_aunit.py:295]: Does it make sense to keep `status` attribute in xml output?

**Review by mnezerka — APPROVED** [review]
Looks good. Just minor findings (rather questions).

File: sap/adt/aunit.py (hunk 2 above)
Comment (jfilak) [reply sap/adt/aunit.py:312]: As mentioned in the commit message:
`I have decided to consider skipped only the tests with the kind ABORTION and the severity TOLERABLE - I consulted ABAP code of cl_abap_unit_assert=>skip( ).`

File: test/unit/test_sap_cli_aunit.py (hunk 3 above)
Comment (jfilak) [reply test/unit/test_sap_cli_aunit.py:295]: The attribute `status` is not banned and it's already there and perhaps it works for somebody.

File: sap/adt/aunit.py (hunk 1 above)
Comment (jfilak) [reply sap/adt/aunit.py:307]: Ah, I know. The class is Alert, the method says if it is an alert saying something was skipped. But the alert itsefl is not skipped. Using the name "is_skipped" would look like that the alert was skipped. Does that make sense?

File: sap/adt/aunit.py (hunk 2 above)
Comment (jfilak) [reply sap/adt/aunit.py:312]: Also added:
```
To be defensive: I would rather receive a complain that we do not 
recognize skipped test than that we missed a real error and incorrectly
marked it as skipped.
```

Comment (mnezerka) [reply sap/adt/aunit.py:312]: Makes sense. Thank's.

File: sap/adt/aunit.py (hunk 1 above)
Comment (mnezerka) [reply sap/adt/aunit.py:307]: It's minor thing, I just saw `is_skipp`, so i tought it should be `is_skipped` or `is_skip`.

Comment (jfilak) [reply sap/adt/aunit.py:307]: :D - uh, oh, eh - I didn't notice the typo ..

**Pull request merged**